### DISCUSSION TITLE
How to store 100% with `TrancheWithPercentage`?

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/v2-periphery#310] [Ideas]
[TrancheWithPercentage](https://github.com/sablier-labs/v2-periphery/blob/staging/src/types/DataTypes.sol#L120C1-L124C6) struct is used to store percentage and tranche duration of a LT stream in case of MLLT.

```solidity
struct TrancheWithPercentage {
    // slot 0
    UD2x18 unlockPercentage;
    uint40 duration;
}
```

However, [UD2x18](https://github.com/PaulRBerg/prb-math/blob/main/src/ud2x18/ValueType.sol) can only have upto 2 digits that means it cannot store 100%. The maximum it can have is 99.9999999....(18 digits).

There could be a use case where there is a single tranche with 100% of the amount.

There are two possible solutions I can think of:
1. Change the datatype to `UD60x18` but it will add a new storage slot.
2. We define a new fixed point number in prb-math such as 4 digits and 14 decimals.

cc @andreivladbrg @PaulRBerg 

PS: I might be wrong with the claim that it cannot take 100%.

**Comment by andreivladbrg** [comment]
`UD2x18` is a `uint64` and the maximum number that `uint64` can have is 2^64, which has [this value](https://www.wolframalpha.com/input/?i=2**64) - a 20-decimal number. This is not a problem because we store the percentage as 1e18 = 100%:

https://github.com/sablier-labs/v2-periphery/blob/8683889a3db3fb2b607861ba0f20b53809710e1f/src/types/DataTypes.sol#L115-L117

Thus, if there is a single tranche, 100% can still be achieved.

**Reply by smol-ninja** [reply]
Yes you are right.
